jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet

```diff
@@ -1,138 +1,137 @@
-local kubeRbacProxyContainer = import '../kube-rbac-proxy/containerMixin.libsonnet';
-local ksm = import 'github.com/kubernetes/kube-state-metrics/jsonnet/kube-state-metrics/kube-state-metrics.libsonnet';
+local krp = (import '../kube-rbac-proxy/container.libsonnet');
 
-{
-  _config+:: {
-    versions+:: {
-      kubeStateMetrics: '1.9.7',
-    },
-    imageRepos+:: {
-      kubeStateMetrics: 'quay.io/coreos/kube-state-metrics',
-    },
-    kubeStateMetrics+:: {
-      scrapeInterval: '30s',
-      scrapeTimeout: '30s',
-      labels: {
-        'app.kubernetes.io/name': 'kube-state-metrics',
-        'app.kubernetes.io/version': $._config.versions.kubeStateMetrics,
-        'app.kubernetes.io/component': 'exporter',
-        'app.kubernetes.io/part-of': 'kube-prometheus',
-      },
-      selectorLabels: {
-        [labelName]: $._config.kubeStateMetrics.labels[labelName]
-        for labelName in std.objectFields($._config.kubeStateMetrics.labels)
-        if !std.setMember(labelName, ['app.kubernetes.io/version'])
-      },
+local defaults = {
+  local defaults = self,
+  namespace: error 'must provide namespace',
+  version: error 'must provide version',
+  image: error 'must provide version',
+  resources: {
+    // TODO(paulfantom): Resource requests should probably be lower.
+    // After checking 2 clusters (one with smaller load and one with heavy load) I got the following stable results:
+    // cluster1 (small load, ksm v2)    : cpu: 5m, memory: 29Mi
+    // cluster2 (heavy load, ksm v1.9.7): cpu: 8m, memory: 190Mi
+    requests: { cpu: '100m', memory: '150Mi' },
+    limits: { cpu: '100m', memory: '150Mi' },
```

Comment:
Why not set it then? :upside_down_face: 
There's always some back and forth with these... :grin: In the end here they are mostly recommendations.

Reply:
I don't have any good values for limits so I didn't want to break more stuff :D

Reply:
These limits haven't been here before this PR have they? It's also valid to do a `resources: {}`.

Reply:
Adjusted resource requests and limits. We can scope it more based on user reports (if there will be any).